src/realmd/AuthSocket.cpp

```diff
@@ -182,7 +189,7 @@ std::array<uint8, 16> VersionChallenge = { { 0xBA, 0xA3, 0x1E, 0x99, 0xA0, 0x0B,
 
 /// Constructor - set the N and g values for SRP6
 AuthSocket::AuthSocket(boost::asio::io_service& service, std::function<void (Socket*)> closeHandler)
-    : Socket(service, std::move(closeHandler)), _status(STATUS_CHALLENGE), _build(0), _accountSecurityLevel(SEC_PLAYER)
+    : Socket(service, std::move(closeHandler)), _status(STATUS_CHALLENGE), promptPin(false), _build(0), _accountSecurityLevel(SEC_PLAYER)
```

Comment:
In the header the member promptPin is declared before _status, so this will result in compiler warnings about members being initialized in an order that doesn't match the declared order (that's also what actually happens, even though the initializer list order suggests otherwise). 

Please re-order them here, or even better, use C++11 and remove promptPin in the initializer list here and declare the member as ```bool promptPin { false };``` for example.

Reply:
I got no compiler warnings on my end, I guess the VS c++ compiler doesn't care much.

So solution is just to initialize promptPin before _status? or are there other members being initialized in wrong order compared to header declaration order?

Reply:
Indeed, the VS compiler doesn't warn about that by default. In fact, I only recently found out it can do so after all, but only in pretty recent versions according to [Microsoft's documentation](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/c5038?view=vs-2017) ("This warning is new in Visual Studio 2017 version 15.3, and is off by default."). But GCC and clang do warn about it for ages, to avoid subtle bugs like one member being initialized with a value based on another uninitialized member, although the order in the code seems to suggest something else. In the case here that's not a problem of course and the code is just fine, but the warnings get annoying.

And yes, just swap the order and initialize promptPin before _status, so it matches the order members are declared in the header, and all is perfect.

Reply:
I did it the other way around, switched places of them in the header instead, thought that made more sense. I am also gonna squash his 2 commits as they touch on the same code making reviewing them a bit messy.